libcudacxx/include/cuda/__barrier/barrier_block_scope.h

```diff
@@ -179,44 +179,21 @@ class barrier<thread_scope_block, ::cuda::std::__empty_completion> : public __bl
   }
 
 private:
-  _CCCL_API inline bool __test_wait_sm_80([[maybe_unused]] arrival_token __token) const
-  {
-    int32_t __ready = 0;
-    NV_DISPATCH_TARGET(
-      NV_PROVIDES_SM_80,
-      (asm volatile(
-         "{\n\t"
-         ".reg .pred p;\n\t"
-         "mbarrier.test_wait.shared.b64 p, [%1], %2;\n\t"
-         "selp.b32 %0, 1, 0, p;\n\t"
-         "}" : "=r"(__ready) : "r"(static_cast<::cuda::std::uint32_t>(::__cvta_generic_to_shared(&__barrier))),
-         "l"(__token) : "memory");))
-    return __ready;
-  }
-
   // Document de drop > uint32_t for __nanosec on public for APIs
   _CCCL_API inline bool __try_wait([[maybe_unused]] arrival_token __token) const
   {
     NV_DISPATCH_TARGET(
       NV_PROVIDES_SM_90,
       (
-        int32_t __ready = 0;
         if (!::cuda::device::is_object_from(__barrier, ::cuda::device::address_space::cluster_shared)) {
           return ::cuda::std::__call_try_wait(__barrier, ::cuda::std::move(__token));
         } else if (!::cuda::device::is_object_from(__barrier, ::cuda::device::address_space::shared)) {
           ::__trap();
-        } asm volatile("{\n\t"
-                       ".reg .pred p;\n\t"
-                       "mbarrier.try_wait.shared.b64 p, [%1], %2;\n\t"
-                       "selp.b32 %0, 1, 0, p;\n\t"
-                       "}" : "=r"(__ready) : "r"(
-                         static_cast<::cuda::std::uint32_t>(::__cvta_generic_to_shared(&__barrier))),
-                       "l"(__token) : "memory");
-        return __ready;),
+        } return ::cuda::ptx::mbarrier_try_wait(__native_handle(), __token);),
```

Comment:
Wound an else here help?
```suggestion
        } 
        else
        {
          return ::cuda::ptx::mbarrier_try_wait(__native_handle(), __token);
        }),
```